packages/wrangler/src/cfetch/internal.ts

```diff
@@ -27,10 +27,11 @@ export async function fetchInternal<ResponseType>(
   resource: string,
   init: RequestInit = {},
   queryParams?: URLSearchParams,
-  abortSignal?: AbortSignal
+  abortSignal?: AbortSignal,
+  authOverride?: string
 ): Promise<ResponseType> {
   await requireLoggedIn();
-  const apiToken = requireApiToken();
+  const apiToken = authOverride || requireApiToken();
```

Comment:
Needed this because we can't pass an `init` with an `Authorization` header already: https://github.com/cloudflare/wrangler2/blob/d691a870e4131741f6480ee12c9024d0ccf0d2c0/packages/wrangler/src/cfetch/internal.ts#L104

If we relaxed that constraint we could back out a bunch of those changes. Thoughts?

Reply:
@petebacondarwin interested in your take on this...